build-tools/webpack/extract-manifest-plugin.js

```diff
@@ -54,79 +54,76 @@ class ExtractManifestPlugin {
+			compilation.hooks.assetPath.tap( PLUGIN_NAME, ( filenameExpression, options, assetInfo ) => {
+				// When there is no `chunk.hashWithLength`, this hook has been called to generate the filename of an asset, not an expression.
+				// Return whatever Webpack generates unmodified.
+				if ( ! ( options.chunk && options.chunk.hashWithLength ) ) return filenameExpression;
 
-					// Check we are not calling ourselfs recursively
-					if ( options.skipExtractManifestPlugin ) return filenameExpression;
+				// Check we are not calling ourselfs recursively
+				if ( options.skipExtractManifestPlugin ) return filenameExpression;
 
-					// Capture the initial quote to make sure we are using the right one when concatenating code.
-					const [ , quote = '"' ] = filenameExpression.match( quoteRegex );
+				// Capture the initial quote to make sure we are using the right one when concatenating code.
+				const [ , quote = '"' ] = filenameExpression.match( quoteRegex );
 
-					// For each variable in brackets, generate a map chunkId->value for that variable and store it in a global function
-					return filenameExpression.replace( variableRegex, function ( match, variable ) {
-						// If the variable has been already captured, don't try to generate the value again
-						if ( ! variablesProcessed.has( variable ) ) {
-							// Call webpack again to generate the chunkId->value map for that specific variable.
-							let nameExpression;
-							try {
-								nameExpression = compilation.mainTemplate.getAssetPath(
-									`"[${ variable }]"`,
-									{
-										...options,
-										// Avoid infinite loops
-										skipExtractManifestPlugin: true,
-									},
-									assetInfo
-								);
-							} catch {
-								// Error trying to get webpack to give us the replacement map. Return the original string
-								// without any replacement and let wepback continue doing its work. The worst case secenario is that we
-								// don't replace this variable with a global function, and therefore the chunkId->value map is embeded
-								// directly in the runtime.
-								return match;
-							}
-
-							// `nameExpression` is an expression that assumes `chunkId` exists in the scope (like `({1:'chunk1', 2:'chunk2'}[chunkId])`).
-							// Transform it to a function and add it to the manifest.
-							manifestContent.push(
-								`${ globalManifest }.set('${ variable }', function(chunkId){return ${ nameExpression };});`
+				// For each variable in brackets, generate a map chunkId->value for that variable and store it in a global function
+				return filenameExpression.replace( variableRegex, function ( match, variable ) {
+					// If the variable has been already captured, don't try to generate the value again
+					if ( ! variablesProcessed.has( variable ) ) {
+						// Call webpack again to generate the chunkId->value map for that specific variable.
+						let nameExpression;
+						try {
+							nameExpression = compilation.getAssetPath(
```

Comment:
https://webpack.js.org/blog/2020-10-10-webpack-5-release/#mainchunkmoduletemplate-deprecation

Method moved to `Compilation`